integration-tests/utils/lyle-utils.js

```diff
@@ -19,13 +19,24 @@ const fetchLyle = async (path, email) => {
   const url = `${lyleUrl}/api/${path}`
   const authClient = await makeAuthClient()
 
-  const res = await fetch(url, {
+  return await fetch(url, {
     method: 'POST',
     headers: { 'Content-Type': 'application/json', ...(await authClient.getRequestHeaders(url)) },
     body: JSON.stringify({ email })
   })
-
-  return res.json()
+    .then(resp => {
+      if (resp.ok) {
+        // response.status >= 200 && response.status < 300
+        return resp.json()
+      }
+      throw Error(`fetch Lyle response: ${resp.text()}`)
+    })
+    .catch(async err => {
+      console.error(err)
+      const errorBody = await err.response.text()
+      console.error(`fetch Lyle request response body: ${errorBody}`)
+      throw err
+    })
```

Comment:
Trying to understand the new logic: Do we need to `throw`, `catch` and then `throw` again? Would something like the following accomplish the same?

```javascript
const fetchLyle = async (path, email) => {
  const url = `${lyleUrl}/api/${path}`
  const authClient = await makeAuthClient()

  const res = await fetch(url, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json', ...(await authClient.getRequestHeaders(url)) },
    body: JSON.stringify({ email })
  })

  res.then(resp => {
    if (resp.ok) {
      // response.status >= 200 && response.status < 300
      return resp.json()
    } else {
      const msg = `fetch Lyle response: ${resp.text()}`
      console.error(msg)
      throw Error(msg)
    }
  })
}
```

Reply:
Oops! I didn't realize throw in `then` block will be caught by `catch` block. I fixed problem.